replica/table.cc

```diff
@@ -2472,6 +2473,7 @@ future<> compaction_group::stop(sstring reason) noexcept {
     auto closed_gate_fut = _async_gate.close();
```

Comment:
I think we have to wait for this gate for correctness since it awaits for pending writes. And we don't want to lose a write. Think for example of merge, that we want to guarantee each stopped group will have all its data (including pending writes) written into sstables.